### PR TITLE
keeping version details in standard version.py file

### PR DESCRIPTION
**Pull request by prashantramangupta**
Resolves issue #196
1. We are removing __version__ key from __init__.py and keeping it in separate file version.py.
2. The code build process (managed in AWS) will update the version in version.py on every release.

### CODE REVIEW

**Comment by raamb** [comment]
> The code release will update the `version.py` here on Github too?

Yes, we will have code build update the version in Github. Since that's an external script wanted to restrict it to a separate file.

> I did this (put it inside `__init__.py`) following modules like [Flask/setup.py](https://github.com/pallets/flask/blob/master/setup.py) and [AWS-CLI/setup.py](https://github.com/aws/aws-cli/blob/develop/setup.py)
> 
> I'd like to suggest to keep it like it is and just increment the version when we improve something that worth it.
Refer to https://packaging.python.org/guides/single-sourcing-package-version/, technique 6 refers to the approach that you mention but it has a potential downside. Keeping the version in its own file makes it simple and easy for the build process to update it during every release (techniques 1 thru 5 are on these lines)

**Comment by arturgontijo** [comment]
> Yes, we will have code build update the version in Github. Since that's an external script wanted to restrict it to a separate file.

So it makes sense.
I'd like to suggest 2 things:

1 - Move `version.py` to the package `snet_cli` folder. (in `setup.py` we can use `from snet_cli.version import __version__`)
2 - Import it in `__init__.py` too, so the users can get the version like:
```
>>> import snet_cli
>>> snet_cli.__version__
'0.2.10'
```

File: setup.py [hunk 1]
```diff
@@ -15,21 +14,27 @@ def install_and_compile_proto():
...
     """Post-installation for installation mode."""
+
     def run(self):
         _install.run(self)
         self.execute(install_and_compile_proto, (), msg="Compile protocol buffers")
 
 
-with open('snet_cli/__init__.py', 'rt', encoding='utf8') as f:
-    version = re.search(r'__version__ = "(.*?)"', f.read()).group(1)
+version = {}
```
Comment (astroseger) [inline setup.py]: 1. What is the reason to set version to ```{}``` at line 34???
2. Is using ```exec``` is necessary here? We could use ```import``` here.

```
import snet_cli.version
version = snet_cli.version.__version__
```

Comment (astroseger) [reply setup.py]: Sorry... There is an error in my previous comment. Of course you simple should do as following:

```python
from snet_cli.version import __version__ as version
```

Comment (prashantramangupta) [reply setup.py]: with
`from snet_cli.version import __version__ as version`
1. This from statement causes the ./snet_cli/__init__.py to be invoked
2. This causes a build failure as some of the dependencies would not have been set up when setup.py is run. This caused the circleci build to fail

Comment (prashantramangupta) [reply setup.py]: @astroseger Can you approve the pr if you are ok?

**Comment by arturgontijo** [comment]
Why we can't have this?

> 2 - Import it in `__init__.py` too, so the users can get the version like:
> ```
> >>> import snet_cli
> >>> snet_cli.__version__
> '0.2.10'
> ```

**Comment by raamb** [comment]
@arturgontijo apologies missed your comment. We can do this, but is there any reason to have it?

**Comment by arturgontijo** [comment]
It is a good practice for python modules:

OpenCV
```
>>> import cv2
>>> cv2.__version__
'3.4.2'
```

AWS-Cli
```
>>> import awscli
>>> awscli.__version__
'1.16.89'
```

But it's just a suggestion, it isn't so important.

**Pull request merged**